src/utilities/tileContentType.js

```diff
@@ -0,0 +1,46 @@
+
+export const TileContentType = {
+	B3DM: 'b3dm',
+	I3DM: 'i3dm',
+	PNTS: 'pnts',
+	CMPT: 'cmpt',
+
+	// 3DTILES_content_gltf
+	GLB: 'glb',
+	GLTF: 'gltf'
+};
+
+// first four bytes ('magic bytes') in big-endian uint32-format
+const MAGIC_BYTES = {
+	0x6233646d: TileContentType.B3DM, // 'b3dm'
+	0x6933646d: TileContentType.I3DM, // 'i3dm'
+	0x706e7473: TileContentType.PNTS, // 'pnts'
+	0x636d7074: TileContentType.CMPT, // 'cmpt'
+	0x676c5446: TileContentType.GLB // 'glTF'
+};
```

Comment:
I think this is a lot of indirection and in the long run won't let users support their own formats with magic bytes like PLY, etc if they choose to use them. I think we can just convert the magic bytes to a string and pass that into the function.

Something like so (untested):

```js
function readMagicBytes( buffer ) {

  const view = new DataView( buffer );
  if ( String.fromCharCode( view.getUint8( 0 ) ) === '{' ) {

    return null;

  }

  let magicBytes = '';
  for ( let i = 0; i < 4; i ++ ) {

    res += String.fromCharCode( view.getUint8( i ) );

  }

  return magicBytes;

}
```

If the file seems to have no magic bytes (JSON) then maybe passing `null` in is reasonable? Then the user can figure out what to do when overriding the function. By default we can assume glTF or fall back to the file extension.

Reply:
Alternatively we could just read the magic bytes in the `parseTile` function itself since the buffer is passed in which would be even simpler. That might have my vote actually.

Reply:
Yeah, I can totally see your point and it wouldn't make much of a difference for what I'm doing. I'll give the last option a go then.